be/src/exprs/CMakeLists.txt

```diff
@@ -25,52 +25,60 @@ set(LIBRARY_OUTPUT_PATH "${BUILD_DIR}/src/exprs")
 # where to put generated binaries
 set(EXECUTABLE_OUTPUT_PATH "${BUILD_DIR}/src/exprs")
 
-add_library(Exprs
+set(EXPR_FILES 
   agg/aggregate_factory.cpp
-  table_function/table_function_factory.cpp
+  anyval_util.cpp
   base64.cpp
-  expr.cpp
   expr_context.cpp
+  expr.cpp
   slot_ref.cpp
-  anyval_util.cpp
+  table_function/table_function_factory.cpp
   vectorized/arithmetic_expr.cpp
-  vectorized/array_expr.cpp
   vectorized/array_element_expr.cpp
+  vectorized/array_expr.cpp
   vectorized/array_functions.cpp
-  vectorized/compound_predicate.cpp
   vectorized/binary_predicate.cpp
-  vectorized/literal.cpp
+  vectorized/bitmap_functions.cpp
+  vectorized/case_expr.cpp
   vectorized/cast_expr.cpp
+  vectorized/column_ref.cpp
+  vectorized/compound_predicate.cpp
+  vectorized/condition_expr.cpp
+  vectorized/encryption_functions.cpp
+  vectorized/es_functions.cpp
+  vectorized/find_in_set.cpp
   vectorized/function_call_expr.cpp
   vectorized/function_helper.cpp
-  vectorized/math_functions.cpp
-  vectorized/string_functions.cpp
-  vectorized/time_functions.cpp
-  vectorized/condition_expr.cpp
-  vectorized/like_predicate.cpp
+  vectorized/geo_functions.cpp
+  vectorized/grouping_sets_functions.cpp
+  vectorized/hyperloglog_functions.cpp
+  vectorized/in_const_predicate.cpp
+  vectorized/info_func.cpp
   vectorized/in_predicate.cpp
-  vectorized/case_expr.cpp
   vectorized/is_null_predicate.cpp
-  vectorized/hyperloglog_functions.cpp
-  vectorized/bitmap_functions.cpp
   vectorized/json_functions.cpp
-  vectorized/percentile_functions.cpp
-  vectorized/encryption_functions.cpp
-  vectorized/geo_functions.cpp
-  vectorized/find_in_set.cpp
+  vectorized/like_predicate.cpp
+  vectorized/literal.cpp
   vectorized/locate.cpp
+  vectorized/math_functions.cpp
+  vectorized/percentile_functions.cpp
+  vectorized/runtime_filter_bank.cpp
+  vectorized/runtime_filter.cpp
   vectorized/split.cpp
   vectorized/split_part.cpp
-  vectorized/column_ref.cpp
-  vectorized/grouping_sets_functions.cpp
-  vectorized/es_functions.cpp
+  vectorized/string_functions.cpp
+  vectorized/time_functions.cpp
   vectorized/utility_functions.cpp
-  vectorized/info_func.cpp
-  vectorized/runtime_filter.cpp
-  vectorized/runtime_filter_bank.cpp
-  vectorized/in_const_predicate.cpp
 )
 
+IF (WITH_HDFS)
```

Comment:
why depends on `with_hdfs`?

Reply:
1. one thread can not attach to more than one virtual machine
2. libhdfs depends on this function and does some initialization,
if the JVM has already created it, it won't create it anymore.

Reply:
I don't understand, we enable java UDF must depend on libhdfs?

Reply:
> A pthread thread can only attach a Java virtual machine. However, libhdfs will create Java virtual machine in the internal, will do some additional initialization of the operation. If we create the Java virtual machine in advance, then these initialization operations will not be executed, resulting in some errors outside of expectations. So we can not create additional virtual machine, need to reuse the virtual machine in libhdfs.

We can additionally implement the getJNIEnv function later to avoid relying on libhdfs.
But if UDF wants to work with libhdfs, you must rely on the functions in libhdfs

@kangkaisen